lighthouse-core/audits/preload-lcp-image.js

```diff
@@ -110,10 +110,64 @@ class PreloadLCPImageAudit extends Audit {
    * @return {{wastedMs: number, results: Array<{url: string, wastedMs: number}>}}
    */
   static computeWasteWithGraph(lcpNode, graph, simulator) {
-    const simulation = simulator.simulate(graph, {flexibleOrdering: true});
-    // For now, we are simply using the duration of the LCP image request as the wastedMS value
-    const lcpTiming = simulation.nodeTimings.get(lcpNode);
-    const wastedMs = lcpTiming && lcpTiming.duration || 0;
+    const simulationBeforeChanges = simulator.simulate(graph, {flexibleOrdering: true});
+    const modifiedGraph = graph.cloneWithRelationships();
+
+    // Store the IDs of the LCP Node's dependencies for later
+    /** @type {Set<string>} */
+    const dependenciesIds = new Set();
+    for (const node of lcpNode.getDependencies()) {
+      dependenciesIds.add(node.id);
+    }
+
+    /** @type {LH.Gatherer.Simulation.GraphNode|null} */
+    let modifiedLCPNode = null;
+    /** @type {LH.Gatherer.Simulation.GraphNode|null} */
+    let mainDocumentNode = null;
+    modifiedGraph.traverse(node => {
+      if (node.type !== 'network') return;
+
+      const networkNode = /** @type {LH.Gatherer.Simulation.GraphNetworkNode} */ (node);
+      if (node.isMainDocument()) {
+        mainDocumentNode = networkNode;
+      } else if (networkNode.record.url === lcpNode.record.url) {
+        modifiedLCPNode = networkNode;
+      }
+    });
+
+    if (!mainDocumentNode) {
+      // Should always find the main document node
+      throw new Error('Could not find main document node');
+    }
+
+    if (!modifiedLCPNode) {
+      // Should always find the LCP node as well or else this function wouldn't have been called
+      throw new Error('Could not find the LCP node');
+    }
+
+    modifiedLCPNode.removeAllDependencies();
+    modifiedLCPNode.addDependency(mainDocumentNode);
+
+    const simulationAfterChanges = simulator.simulate(modifiedGraph, {flexibleOrdering: true});
+    const lcpTimingsBefore = simulationBeforeChanges.nodeTimings.get(lcpNode);
+    const lcpTimingsAfter = simulationAfterChanges.nodeTimings.get(modifiedLCPNode);
+    /** @type {Map<String, LH.Gatherer.Simulation.GraphNode>} */
+    const modifiedNodesById = Array.from(simulationAfterChanges.nodeTimings.keys())
+      .reduce((map, node) => map.set(node.id, node), new Map());
+    const lcpEndTimeBefore = lcpTimingsBefore && lcpTimingsBefore.endTime || 0;
```

Comment:
should we throw if either of these times are undefined? or at least mark notApplicable with a warning?

if it's impossible, then a `throw new Error('Impossible - LCP node always exists by this point')`?